server/engine/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/cbl/CblParser.java

```diff
@@ -0,0 +1,444 @@
+    removeSegments(start, pos);
+  }
+
+  private void removePrevComma(int from, int to) {
+    for (int i = to - 1; i >= from; i--) {
+      if (segments[i] == null || isBlank(segments[i])) {
+        continue;
+      }
+      if (",".equals(segments[i])) {
+        segments[i] = null;
+      }
+      return;
+    }
+  }
+
+  private void removeSegments(int start, int end) {
+    Arrays.fill(segments, start, end, null);
+  }
+
+  private CompilerDirectiveNode createDirectiveNode(int start, int end) {
+    StringBuilder text = new StringBuilder();
+    for (int i = start; i < end; i++) {
+      text.append(segments[i].trim());
+    }
+    Range range =
+        new Range(new Position(lineNumber, ranges[start]), new Position(lineNumber, ranges[end]));
+    Locality locality = Locality.builder().uri(uri).range(range).build();
+    return new CompilerDirectiveNode(locality, text.toString(), CICSDialect.DIALECT_NAME);
+  }
+
+  private void semanticError(String next, String message) {
+    diagnostics.add(
+        new CblDiagnosticException(makeCurrentLocation(), next, WARNING, message)
+            .toSyntaxError(lineNumber));
+  }
+
+  private boolean isLineNumberWrong(int nnn) {
+    // nnn must be an integer between 1 and 255.
+    if (nnn == 0) {
+      return true;
+    }
+    return nnn <= 0 || nnn > 255;
+  }
+
+  private int checkIntegerLiteral(String next) throws CblDiagnosticException {
+    try {
+      return Integer.parseInt(next);
+    } catch (NumberFormatException e) {
+      throw expect("integer", makeCurrentLocation(), "integer");
+    }
+  }
+
+  private Location makeCurrentLocation() {
+    Position start = new Position(lineNumber, ranges[pos - 1]);
+    Position end = new Position(lineNumber, ranges[pos]);
+    Range range = new Range(start, end);
+    return new Location(uri, range);
+  }
+
+  private void parseOption() throws CblDiagnosticException {
+    int depth = 0;
+    while (hasMore()) {
+      if ("'".equals(peek())) {
+        skipAfter("'");
+        continue;
+      }
+      if ("\"".equals(peek())) {
+        skipAfter("\"");
+        continue;
+      }
+
+      if (depth == 0)
+        if (isNext(")")) {
+          throw expect(next(), makeCurrentLocation());
+        } else if (isNext(",")) {
+          break;
+        }
+      String next = next();
+      if (next.equalsIgnoreCase("(")) {
+        depth++;
+      } else if (next.equalsIgnoreCase(")")) {
+        depth--;
+      }
+    }
+    if (depth > 0) {
+      throw expect(peek(), makeCurrentLocation(), ")");
+    }
+    opt(",");
+  }
+
+  private boolean isNext(String expected) {
+    return hasMore() && peek().equalsIgnoreCase(expected);
+  }
+
+  private void or(String... variants) throws CblDiagnosticException {
+    String token = next();
+    for (String text : variants) {
+      if (text.equalsIgnoreCase(token)) {
+        return;
+      }
+    }
+    throw expect(token, makeCurrentLocation(), variants);
+  }
```

Comment:
```suggestion
  private int or(String... variants) throws CblDiagnosticException {
    String token = next();
    for (String text : variants) {
      if (text.equalsIgnoreCase(token)) {
        return pos - 1;
      }
    }
    throw expect(token, makeCurrentLocation(), variants);
  }
```
the `one` could return the index as well so that you can write something like
```java
    int start = or("FLAG", "F");
```